src/__tests__/header.test.jsx

```diff
@@ -0,0 +1,83 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+import { act } from '@testing-library/react-hooks';
+import ThemeState from '../context/theme/themeState';
+import VideoState from '../context/video/VideoState';
+import Header from '../components/Header';
+
+import videosMock from '../mocks/youtube-videos-mock.json';
+import axiosClient from '../config/axios';
+
+jest.mock('../config/axios');
+describe('Header component tests', () => {
+  beforeEach(async () => {
+    axiosClient.mockReturnValue(
+      Promise.resolve({
+        data: { ...videosMock },
+      })
+    );
+    await act(async () => {
+      render(
+        <VideoState>
+          <ThemeState>
+            <Header />
+          </ThemeState>
+        </VideoState>
+      );
+    });
+  });
+
+  test('should containt a toggle button', () => {
+    const toggle = screen.getByRole('checkbox');
+    expect(toggle).toBeInTheDocument();
+  });
+
+  test('Toggle button should be checkbox type', () => {
+    const toggle = screen.getByRole('checkbox');
+    expect(toggle).toHaveAttribute('type', 'checkbox');
```

Comment:
Since you are doing the query `getByRole('checkbox')`, there is explicitly that the element is a checkbox, so you could do the expect using `toBeInTheDocument()`, makes sense?